x-pack/elastic-agent/pkg/agent/transpiler/steps.go

```diff
@@ -212,6 +217,50 @@ func MoveFile(path, target string, failOnMissing bool) *MoveFileStep {
 	}
 }
 
+// ExecFileStep executes a file.
+type ExecFileStep struct {
+	Path string
+	Args []string
+}
+
+// Execute executes file with provided arguments.
+func (r *ExecFileStep) Execute(rootDir string) error {
+	path, isSubpath, err := joinPaths(rootDir, r.Path)
+	if err != nil {
+		return err
+	}
+
+	if !isSubpath {
+		return fmt.Errorf("invalid path value for operation 'Exec': %s", path)
+	}
+
+	args := []string{path}
+	args = append(args, r.Args...)
+	cmd := &exec.Cmd{
+		Path: path,
+		Args: args,
+		Env:  nil,
+		Dir:  rootDir,
+	}
+	output, err := cmd.Output()
```

Comment:
if things goes wrong here and exec hangs forever we're basically done without a way for recover. maybe we could use wait + timeout and after that checking output (redirecting it to byte array)

Reply:
I will update this to have a timeout, good idea!

Reply:
@michalpristas Latest code has timeout implemented in `exec_file`.